component/grant-type/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/TokenExchangeGrantHandler.java

```diff
@@ -0,0 +1,272 @@
+        super.init();
+
+        /*
+          From identity.xml following configs are read.
+
+          <OAuth>
+              <TokenExchangeGrant>
+                  <EnableIATValidation>true</EnableIATValidation>
+                  <IATValidityPeriod>30</IATValidityPeriod>
+              </TokenExchangeGrant>
+          </OAuth>
+         */
+
+        String validateIATProp = IdentityUtil.getProperty(TokenExchangeConstants.PROP_ENABLE_IAT_VALIDATION);
+        if (StringUtils.isNotBlank(validateIATProp)) {
+            validateIAT = Boolean.parseBoolean(validateIATProp);
+        }
+
+        String validityPeriodProp = IdentityUtil.getProperty(TokenExchangeConstants.PROP_IAT_VALIDITY_PERIOD);
+
+        if (validateIAT) {
+            if (StringUtils.isNotBlank(validityPeriodProp)) {
+                try {
+                    validityPeriod = Integer.parseInt(validityPeriodProp);
+                } catch (NumberFormatException e) {
+                    validityPeriod = TokenExchangeConstants.DEFAULT_IAT_VALIDITY_PERIOD;
+                    log.warn("Invalid value: " + validityPeriodProp + " is set for IAT validity period. Using " +
+                            "default value: " + validityPeriod + " minutes.");
+                }
+            } else {
+                validityPeriod = TokenExchangeConstants.DEFAULT_IAT_VALIDITY_PERIOD;
+                log.warn("Empty value is set for IAT validity period. Using default value: " + validityPeriod
+                        + " minutes.");
+            }
+        }
+
+        String registeredClaims = IdentityUtil.getProperty(TokenExchangeConstants.REGISTERED_CLAIMS);
+        if (StringUtils.isNotBlank(registeredClaims)) {
+            registeredClaimNames = registeredClaims.split("\\s*,\\s*");
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Validate IAT is set to: " + validateIAT + " for JWT grant.");
+            if (validateIAT) {
+                log.debug("IAT validity period is set to: " + validityPeriod + " minutes for JWT grant.");
+            }
+        }
+    }
+
+    @Override
+    public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2Exception {
+        SignedJWT signedJWT;
+        IdentityProvider identityProvider;
+        String tokenEndPointAlias;
+        JWTClaimsSet claimsSet = null;
+        String requested_audience = null;
+        boolean audienceFound;
+
+        RequestParameter[] params = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getRequestParameters();
+        Map<String, String[]> requestParams = Arrays.stream(params).collect(Collectors.toMap(RequestParameter::getKey,
+                RequestParameter::getValue));
+        if (requestParams.get(TokenExchangeConstants.REQUESTED_TOKEN_TYPE) != null) {
+            requested_token_type = requestParams.get(TokenExchangeConstants.REQUESTED_TOKEN_TYPE)[0];
+        }
+        if(requestParams.get(TokenExchangeConstants.AUDIENCE) != null) {
```

Comment:
formatting
```suggestion
        if (requestParams.get(TokenExchangeConstants.AUDIENCE) != null) {
```

Reply:
Fixed in 8693d5454cb05015e9d9614c56a1d09bdf8a8997